sdk/core/azure-core/src/main/java/com/azure/core/util/paging/ContinuablePagedIterable.java

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.paging;
+
+import com.azure.core.util.IterableStream;
+import java.util.stream.Stream;
+
+/**
+ * This class provides utility to iterate over {@link ContinuablePage} using {@link Stream}
+ * {@link Iterable} interfaces.
+ *
+ * @param <C> the type of the continuation token
+ * @param <T> The type of elements in a {@link ContinuablePage}
+ * @param <P> The {@link ContinuablePage} holding items of type {@code T}.
+ *
+ * @see IterableStream
+ * @see ContinuablePagedFlux
+ */
+public abstract class ContinuablePagedIterable<C, T, P extends ContinuablePage<C, T>> extends IterableStream<T> {
+    private static final int DEFAULT_BATCH_SIZE = 1;
+    private final ContinuablePagedFlux<C, T, P> pagedFlux;
+
+    /**
+     * Creates instance given {@link ContinuablePagedFlux}.
+     *
+     * @param pagedFlux the paged flux use as iterable
+     */
+    public ContinuablePagedIterable(ContinuablePagedFlux<C, T, P> pagedFlux) {
+        super(pagedFlux);
+        this.pagedFlux = pagedFlux;
+    }
+
+    /**
+     * Retrieve the {@link Stream}, one page at a time.
+     * It will provide same {@link Stream} of T values from starting if called multiple times.
+     *
+     * @return {@link Stream} of a pages
+     */
+    public Stream<P> streamByPage() {
```

Comment:
Adding Javadoc about default batch size would be good for user and improve quality of Javadoc.

Reply:
yes, we will document it as part of adding a ctr argument for the batch size.

Reply:
I've added batchsize based ctr with Javadoc.